### PR TITLE
Rewrite branch logic to allow base branches other than master

### LINKED ISSUE #2: Diff only compares current branch with master
The script only compares current branch with master right now. It could check for closest branch https://stackoverflow.com/questions/3161204/how-to-find-the-nearest-parent-of-a-git-branch but that isn't always correct.
```
git show-branch -a \
| grep '\*' \
| grep -v `git rev-parse --abbrev-ref HEAD` \
| head -n1 \
| sed 's/.*\[\(.*\)\].*/\1/' \
| sed 's/[\^~].*//'
```

The best way is probably using the Github API https://developer.github.com/v3/repos/commits/#list-pull-requests-associated-with-commit
```
GET /repos/:owner/:repo/commits/:commit_sha/pulls
```

### PR DESCRIPTION
**Pull request by jplatte**
As a consequence of how it now works, it can no longer be run on non-`pull_request` events.

Fixes #2.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,42 +1,37 @@
...
-  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin master
-  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --shallow-exclude=master origin ${BRANCH}
+  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin "${BASE_REF}:__ci_base"
+  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --shallow-exclude="${BASE_REF}" origin "${PR_REF}:__ci_pr"
   # Get the list before the "|| true" to fail the script when the git cmd fails.
-  COMMIT_LIST=`/usr/bin/git log --pretty=format:%s origin/master..origin/${BRANCH}`
+  COMMIT_LIST=$(/usr/bin/git log --pretty=format:%s __ci_base..__ci_pr)
 
-  FIXUP_COUNT=`echo $COMMIT_LIST | grep fixup! | wc -l || true`
+  FIXUP_COUNT=$(echo ${COMMIT_LIST} | grep 'fixup!' | wc -l || true)
```
Comment (13rac1) [inline entrypoint.sh]: This reformatting should be a separate PR, separate commit, or not done since it's cluttering the review of the important changes. Are you using a bash-script automatic formatter? What standards are you changing this formatting to?

File: .github/workflows/git.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: Git Checks
 
-on: [push]
+on: [pull_request]
```
Comment (13rac1) [inline .github/workflows/git.yml:3]: According to the documentation and "documentation": https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request
> Note: By default, a workflow only runs when a pull_request's activity type is opened, synchronize, or reopened. To trigger workflows for more activity types, use the types keyword.

https://github.community/t/what-is-a-pull-request-synchronize-event/14784/2
> What this implies is that if you push a new commit to the HEAD ref of a pull request then this “synchronize” event will be triggered. This is because the system is syncing the pull requests with the latest changes. This will NOT trigger if the base ref is updated.

Ah, it is effectively the same for our purposes, but Github's documentation is not clear.

**Review by 13rac1 — CHANGES_REQUESTED** [review]
A question about the formatting, it looks reasonable otherwise. Thank you!

File: .github/workflows/git.yml (hunk 2 above)
Comment (jplatte) [reply .github/workflows/git.yml:3]: Right, I actually noticed this before, it's not new (there's also no `push` events when changing the base of a PR). I guess one would have to add `types: [assigned, opened, synchronize, reopened, edited]` to correctly handle that, although that will also trigger if just the description or title is updated. Should I add that?

File: entrypoint.sh (hunk 1 above)
Comment (jplatte) [reply entrypoint.sh]: Sorry. I'm used to doing stuff like this on the fly since most things I contribute to don't keep their git history clean anyway. Of course that's not the case here 😆

I've updated the PR to keep the formatting as-is. I recommend running shellcheck over the script though, I think backticks are often discouraged and you're also not consistent about `$VAR` vs `${VAR}`. I wasn't going for some specific standard, just more consistency and what I had in my head as best practices.

**Comment by jplatte** [comment]
ping @13rac1

File: .github/workflows/git.yml (hunk 2 above)
Comment (13rac1) [reply .github/workflows/git.yml:3]: 🤔 This _technically_ doesn't need to run on a base ref change, since it only matters if the commits themselves change.

File: entrypoint.sh (hunk 1 above)
Comment (13rac1) [reply entrypoint.sh]: Good point about the consistency. This works for now though.

**Review by 13rac1 — APPROVED** [review]

**Comment by 13rac1** [comment]
Sorry it took a whole year to reply 😳

**Comment by 13rac1** [comment]
@jplatte Can you update the commit message to answer "why?" to "As a consequence of how it now works, it can no longer be run on non-pull_request events."

**Comment by jplatte** [comment]
Is this better?

**Comment by 13rac1** [comment]
Yes! Thank you :D

**Comment by jplatte** [comment]
@13rac1 Mind publishing a new version?
